### DISCUSSION TITLE
LSP support

### DISCUSSION THREAD

**Discussion by ericdallo** [jank-lang/jank#14] [Issue Triage] [answered]
Hi!
Looking forward to this project!

I'd like to discuss if you already started working on anything related to LSP support, I'm maintainer of [clojure-lsp](https://clojure-lsp.io) and have a lot of experience with LSP protocol and editors using it so if you agree I can try to help with that :)

Some questions:
- It'd be nice to implement in jank or clojure IMO, and we could use lsp4clj which clojure-lsp uses under the hood and helps a lot with all the communication process and infra, WDYT?

**Comment by ericdallo** [comment]
AFAICS, it behaves similar to Clojure as 100% compatible, so using clojure-lsp would already work, right?
Is there anything different that a LSP clojure-lsp doesn't support already?

**Comment by jaidetree** [comment]
According to the docs one of the goals is to support static typing, so getting more static type analysis and completions from that would be cool.

**Comment by jeaye** [comment, marked as answer]
Not sure how I missed this before. Your help would be *very* appreciated, Eric! I think clojure-lsp should do well out of the gate, since jank's first big milestone is just parity with Clojure JVM. However, going beyond that, I'd like to have jank provide much more useful information via LSP, both including static type information and semantic information which can be used for semantic highlighting. I'm very open to reusing parts of the compiler to aid in lexing, parsing, and analyzing jank code and I want jank's tooling to feel even better than Clojure's.
